common/src/main/java/com/cssbham/cssminecraft/common/permission/PermissionPluginServiceFactory.java

```diff
@@ -0,0 +1,38 @@
+package com.cssbham.cssminecraft.common.permission;
+
+import java.lang.reflect.InvocationTargetException;
+
+/**
+ * Factory methods for getting a {@link PermissionPluginService}
+ */
+public class PermissionPluginServiceFactory {
+
+    public static PermissionPluginService any() {
+         String[] plugins = new String[]{ "LuckPerms" };
+        for (String plugin : plugins) {
+            try {
+                return forPlugin(plugin);
+            } catch (RuntimeException ignored) { }
+        }
+        return new StubPermissionPluginService();
+    }
+
+    public static PermissionPluginService forPlugin(String plugin) {
+        try {
+            return switch (plugin) {
+                case "LuckPerms" -> checkClassAndBuild("net.luckperms.api.LuckPermsProvider", LuckPermsPermissionPluginService.class);
+                default -> new StubPermissionPluginService();
+            };
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException | NoSuchMethodException |
+                 InvocationTargetException e) {
+            throw new RuntimeException(String.format("Permission plugin %s is not available", plugin));
+        }
+    }
```

Comment:
Would we rather have some kind of registration for permission services rather than hardcoding them?

Reply:
I don't see the case where permission plugins will be registering themselves here. But if in the future we do have 30 different permission plugins we want to integrate with, then this factory could become abstract and a concrete version can be provided at runtime (rather than having an explicit list of supported plugins here)